pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -481,12 +484,11 @@ def __call__(
 
         else:  # We are not on the root process so go into waiting loop:
             self._waitLoop()
-            sol = None
 
         # Communication solution and return
-        sol = self._communicateSolution(sol)
+        commSol = self._communicateSolution(sol)
```

Comment:
I renamed these because `mypy` was getting confused and thinking `sol` could be `None`, so the answer was just to reassign this a new variable name.